libedit/config.h

```diff
@@ -32,7 +32,11 @@
 
 /* Define to 1 if you have getpwnam_r and getpwuid_r that are POSIX.1
    compatible. */
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
```

Comment:
The __unix__ macro is also defined on linux, so I would suggest changing the MacOS detection to
#if defined (defined(__APPLE__) && defined(__MACH__))

Currently it breaks the linux build.